packages/shared/src/components/MainLayout.tsx

```diff
@@ -89,6 +90,19 @@ export default function MainLayout({
   const { trackEvent } = useContext(AnalyticsContext);
   const { sidebarRendered } = useSidebarRendered();
   const [openMobileSidebar, setOpenMobileSidebar] = useState(false);
+  const handlers = useSwipeable({
+    onSwipedLeft: () => {
+      if (!sidebarRendered && openMobileSidebar) {
+        setOpenMobileSidebar(false);
+      }
+    },
+    onSwipedRight: () => {
+      if (!sidebarRendered && !openMobileSidebar) {
+        setOpenMobileSidebar(true);
+      }
+    },
+    preventDefaultTouchmoveEvent: true,
+  });
```

Comment:
This calls for a custom hook 😉

Reply:
Even with a custom hook i'm getting the error that the hook usage is wrong.
Do you have any idea why this occurs?